### DISCUSSION TITLE
Webpack5迁移MSFU 编译失败问题

### DISCUSSION THREAD

**Discussion by congxiaobai** [umijs/umi#10628] [Q&A]
webapck 版本 5.75.0
node版本 16.0.0 | 14.13.0   
MFSU 版本 4.0.52
pnpm 版本 7.27.0

背景：
使用webpack 编译启动正常，按照https://umijs.org/blog/mfsu-independent-usage#4-%E8%AE%BE%E5%AE%9A-webpack-%E9%85%8D%E7%BD%AE 指引，迁移至于MSFU。
本项目是menorepo，结构目录如下：
![image](https://user-images.githubusercontent.com/20089891/221730737-951f9fd4-563e-4326-b56d-2182286f89a4.png)

其中项目erp与foundation,通过Pnpm的workspace建立软连接，进行编译。
目前出现两个情况：
  1. 首次启动，生成了.mfsu目录，但是foundation相关的代码会报错，出现 'react' 'mobx'未找到异常
  2. 删除.msfu目录后，再次启动，无法再次生成.msfu文件，并且也无异常打印

**Comment by congxiaobai** [comment]
![image](https://user-images.githubusercontent.com/20089891/221733601-ea916873-1313-46e3-a203-ec88540554da.png)
初次编译报错如图

**Comment by fz6m** [comment]
需要给最小复现。
